core/vm/contract.go

```diff
@@ -109,6 +111,14 @@ func isCodeFromAnalysis(analysis []uint64, udest uint64) bool {
 // isCode returns true if the provided PC location is an actual opcode, as
 // opposed to a data-segment following a PUSHN operation.
 func (c *Contract) isCode(udest uint64) bool {
+	if !c.IsLegacy() {
+		// TODO remove this
+		panic("shouldn't run jumpdest analysis on EOF")
+	}
+	// Do we already have an analysis laying around?
+	if c.analysis != nil {
```

Comment:
This is superfluous. `return isCodeFromAnalysis(analysis, udest)` does the same already.

Reply:
Thanks @yperbasis. Nice catch.

Resolved by [e30257c](https://github.com/ledgerwatch/erigon/pull/6215/commits/e30257c8811a5eb5ac8c4a3101e1b2f727e7717c)